eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -236,6 +238,24 @@ jobs:
           artifactName: '$(coreClrProductArtifactName)'
           displayName: 'CoreCLR product download for Mono'
 
+
+    # Build target specific tests
+    - ${{ if ne(parameters.corefxTests, true) }}:
+      - ${{ if eq(parameters.testBuildPhased, true) }}:
+        # Install test build dependencies
+        - ${{ if eq(parameters.osGroup, 'OSX') }}:
+          - script: sh $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup)
+            displayName: Install native dependencies
+        - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          # Necessary to install correct cmake version
+          - script: $(Build.SourcesDirectory)\eng\common\init-tools-native.cmd -InstallDirectory $(Build.SourcesDirectory)\native-tools -Force
+            displayName: Install native dependencies
+
+        # Build targetSpecific managed test components
+        - script: $(coreClrRepoRootDir)build-test$(scriptExt) targetSpecific skipnative skipgeneratelayout skiptestwrappers $(buildConfig) $(archType) $(crossArg) $(priorityArg) ci $(librariesOverrideArg)
+          displayName: Build managed test components
+
```

Comment:
The earliest draft of this was also adding an archive of the tests.
I was fiddling with two variants.  
1. Archive all tests
2. Archive target specific tests

I though that if performance was really critical, no artifact would be faster.